### PR TITLE
Adding support for custom base image for creating packer image

### PR DESCRIPTION
**Pull request by bishal-pdMSFT**

### CODE REVIEW

File: Tasks/PackerBuild/task.json [hunk 1]
```diff
@@ -103,18 +103,58 @@
...
+            "type": "pickList",
+            "label": "Base image source",
+            "required": true,
+            "defaultValue": "default",
+            "helpMarkDown": "Select the source for base image.",
+            "properties": {
+                "EditableOptions": "True"
+            },
+            "options": {
+                "default": "Builtin",
```
Comment (sachinma) [inline Tasks/PackerBuild/task.json]: lets name lhs also as builtIn

File: Tasks/PackerBuild/src/builtinTemplateFileProvider.ts [hunk 2]
```diff
@@ -13,8 +13,10 @@ export default class BuiltInTemplateFileProvider extends TemplateFileProviderBas
     constructor() {
```
Comment (sachinma) [inline Tasks/PackerBuild/src/builtinTemplateFileProvider.ts]: Filename builtin can have i capital

File: Tasks/PackerBuild/task.json (hunk 1 above)
Comment (bishal-pdMSFT) [inline Tasks/PackerBuild/task.json]: rhs has been re-named as "Gallery" after review with Atul. And in my opinion, "default" is better than "gallery" from code perspective

File: Tasks/PackerBuild/src/builtinTemplateFileProvider.ts (hunk 2 above)
Comment (bishal-pdMSFT) [inline Tasks/PackerBuild/src/builtinTemplateFileProvider.ts]: done

**Pull request merged**